OpenRA.Mods.Common/Traits/Player/ClassicProductionQueue.cs

```diff
@@ -104,7 +108,7 @@ protected override bool BuildUnit(ActorInfo unit)
 				return false;
 			}
 
-			foreach (var p in producers.Where(p => !p.Actor.IsDisabled()))
+			foreach (var p in producers.Where(p => !p.Trait.IsTraitPaused))
```

Comment:
Minor perf nit: no benefit to using LINQ here, this can move to an `if () continue` at the top of the loop.